envoy/network/io_handle.h

```diff
@@ -65,6 +65,12 @@ class IoHandle {
    */
   virtual bool isOpen() const PURE;
 
+  /**
+   * Return true if connect() has successfully been called on the socket.
+   * Use isOpen() to check if the socket is still connected or not.
```

Comment:
Hm. The naming here seems tricky, though I'm not sure what the best approach is. But the fact that we need to call `isOpen` to check if the socket `is connected` is unfortunate. It makes me think someone is going to end up calling `isConnected()` in the future and be very surprised by the results :)

Maybe instead of `isConnected()` we could use a name like `wasConnected()` or `hasConnected` or something? WDYT?

Reply:
I struggled with this one too.  `wasConnected()` is a good idea - I switched to using that.